Runtime/SpriteSwitchEventInvoker.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Unity.MaterialSwitch
+{
+    [AddComponentMenu("")]
+    [ExecuteAlways]
+    internal class SpriteSwitchEventInvoker : MonoBehaviour
```

Comment:
We have a class in FilmInternalUtilities called `MonoBehaviourSingleton` to create a singleton instance.
This has been used in other packages as well, so maybe it's a good idea to use that here as well ?